pymongo/collection.py

```diff
@@ -954,9 +954,9 @@ def _delete(
             else:
                 delete_doc['collation'] = collation
         if hint is not None:
-            if not acknowledged:
+            if sock_info.max_wire_version < 9:
```

Comment:
A strict reading of the spec (and spec tests) actually requires this to be: `if not acknowledged && sock_info.max_wire_version < 8`:

```
class DeleteOptions {
...
   * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
   * This option is only supported by servers >= 4.4. Older servers >= 3.4 will report an error for using this option.
   * For servers < 3.4, the driver MUST raise an error if the caller explicitly provides a value.
   * For unacknowledged writes using OP_DELETE, the driver MUST raise an error if the caller explicitly provides a value.
   * For unacknowledged writes using OP_MSG and servers < 4.4, the driver MUST raise an error if the caller explicitly provides a value.
   *
   * @see https://docs.mongodb.com/manual/reference/command/delete/
   */
  hint: Optional<(String | Document)>;
```

Reply:
It should be `if not acknowledged && sock_info.max_wire_version < 9`  (driver 4.4), correct?

Reply:
Yes you're right, my mistake.